orchagent/mirrororch.cpp

```diff
@@ -320,91 +316,128 @@ bool MirrorOrch::getNeighborInfo(const string& name, MirrorEntry& session)
 {
     SWSS_LOG_ENTER();
 
-    NeighborEntry neighbor;
-    MacAddress mac;
-
-    assert(session.nexthopInfo.resolved);
-
-    SWSS_LOG_INFO("Getting neighbor info for %s session\n", name.c_str());
-
-    if (!m_neighOrch->getNeighborEntry(session.nexthopInfo.nexthop, neighbor, mac))
+    // 1) Check if the destination IP is directly connected and get neighbor
+    // 2) Check if the session has next hop and neighbor information is available
```

Comment:
rewrite for clarity

1. if session destination IP is directly connected, then get the neighbor info.
2. if session has next hop, then the get neighbor info of the next hop.
3. otherwise, return false.